orcoursetrion/lib/github.py

```diff
@@ -390,3 +393,59 @@ def delete_web_hooks(self, org, repo):
                 raise GitHubUnknownError(response.text)
             num_hooks_removed += 1
         return num_hooks_removed
+
+    def shallow_copy_repo(self, src_repo, dst_repo, committer, branch=None):
+        """Copies one branch repo's contents to a new repo in the same
+        organization without history.
+
+        .. DANGER::
+           This will overwrite the destination repo's default branch and
+           rewrite its history.
+
+        The basic workflow is:
+
+        - Clone source repo
+        - Remove source repo ``.git`` folder
+        - Initialize as new git repo
+        - Set identity
+        - Add everything and commit
+        - Force push to destination repo
+
+        Args:
+            src_repo (str): Full git url to source repo.
+            dst_repo (str): Full git url to destination repo.
+            committer (dict): {'name': ..., 'email': ...} for the name
+                and e-mail to use in the initial commit of the
+                destination repo.
+            branch (str): Option branch, if not specified default is used.
+        Raises:
+            sh.ErrorReturnCode
+        Returns:
+            None
+
+        """
+        # Grab current working directoy so we return after we are done
+        cwd = unicode(sh.pwd().rstrip('\n'))
+        tmp_dir = tempfile.mkdtemp(prefix='orc_git')
+        try:
+            sh.cd(tmp_dir)
+            if branch is None:
+                sh.git.clone(src_repo, depth=1)
+            else:
+                sh.git.clone(src_repo, depth=1, branch=branch)
+            sh.cd(sh.glob('*')[0])
```

Comment:
I suppose it probably would make more sense just to add an arg specifying a known folder in the clone command.  I'll update it to that instead